src/pages/Inventory.page.js

```diff
@@ -3,13 +3,39 @@ import { BaseSwagLabPage } from './BaseSwagLab.page';
 export class InventoryPage extends BaseSwagLabPage {
     url = '/inventory.html';
 
-    headerTitle = this.page.locator('.title');
+    headerTitle = this.page.getByTestId('title');
 
-    inventoryItems = this.page.locator('.inventory_item');
+    inventoryProducts = this.page.getByTestId('inventory-item');
 
-    addItemToCartButton = this.page.locator('[id^="add-to-cart"]');
+    addItemToCartButton = this.page.getByTestId('add-to-cart-sauce-labs-backpack');
+
+    select = this.page.getByTestId('product-sort-container');
+
+    options = this.select.locator('option');
 
     async addItemToCartById(id) {
         await this.addItemToCartButton.nth(id).click();
     }
+
+    async getProductData() {
```

Comment:
getProductsData or getAllProductsData or something like that.
In your method you're collecting data for all products, not just one. So, name your method accordingly

Also, what if you want to grab one product data? 
Maybe it worth creating separate method to do so. And use it in your method when collecting data for all products.